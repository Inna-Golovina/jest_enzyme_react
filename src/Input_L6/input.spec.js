//тестирование событий и обработчиков событий
// в инпуте никакой внутренней логики нет, будет достаточно провести снепшот-тестирование

import { shallow } from "enzyme";
import React from "react";
import Input from "./input";

describe('Input component', () => {
  it('should use default onChange', () => {
    const component = shallow(<Input />);
    expect(component).toMatchSnapshot();
  })

  describe('defaultProps', () => {
    it('should use default onChange', () => { //для метода onChange
      const result = Input.defaultProps.onChange();
      expect(result).toBe(undefined)
    })

    it('should use default onKeyPress', () => { //для метода onKeyPress
      const result = Input.defaultProps.onKeyPress();
      expect(result).toBe(undefined)
    })
  })
})