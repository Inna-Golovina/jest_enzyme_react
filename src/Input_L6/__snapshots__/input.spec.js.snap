// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should use default onChange 1`] = `
<div
  className="inputWrapper"
>
  <i
    className="fas fa-search"
  />
  <input
    className="input"
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="Click to search"
    value=""
  />
</div>
`;
